app/vendor/views.py

```diff
@@ -48,6 +50,53 @@ def new_listing():
     return render_template('vendor/new_listing.html', form=form)
 
 
+@vendor.route('/csv-upload', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def csv_upload():
+    """Create a new item."""
+    form = NewCSVForm()
+    listings = []
+    if form.validate_on_submit():
+        csv_field = form.file_upload
+        buff = csv_field.data.stream
+        csv_data = csv.DictReader(buff, delimiter=',')
+        #for each row in csv, create a listing
+        current_vendor = Vendor.get_vendor_by_user_id(user_id=current_user.id)
```

Comment:
discussed above. generally its a good idea to abstract data handling to model class imo.